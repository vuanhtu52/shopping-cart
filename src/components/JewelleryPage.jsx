const JewelleryPage = () => {
    return <div>Jewellery</div>;
};

export default JewelleryPage;