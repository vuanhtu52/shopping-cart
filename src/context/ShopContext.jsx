import { createContext } from "react";

const ShopContext = createContext({
    products: [],
    cartItems: [],
    addToCart: () => {},
    addMultipleToCart: () => {},
    removeOneFromCart: () => {},
    getTotalCartItems: () => {},
});

export default ShopContext;