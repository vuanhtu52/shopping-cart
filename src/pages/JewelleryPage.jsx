import { useContext } from "react";
import ShopContext from "../context/ShopContext";
import ItemCard from "../components/ItemCard";

const JewelleryPage = () => {
    const { products } = useContext(ShopContext);

    return <div>
        <h2 className="text-center text-xl font-bold p-4">Jewellery</h2>
        <div className="grid gap-8 justify-items-center grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 p-4">
            {
                products
                    .filter(product => product.category === "jewelery")
                    .map(product => <ItemCard key={product.id} item={product} />)
            }
        </div>
    </div>;
};

export default JewelleryPage;