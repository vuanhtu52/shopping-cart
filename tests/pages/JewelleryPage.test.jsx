import { render, screen } from "@testing-library/react";
import ShopContext from "../../src/context/ShopContext";
import { MemoryRouter } from "react-router-dom";
import JewelleryPage from "../../src/pages/JewelleryPage";
import { describe } from "vitest";

const renderComponent = products => {
    return render(
        <ShopContext.Provider value={{ products }}>
            <MemoryRouter>
                <JewelleryPage />
            </MemoryRouter>
        </ShopContext.Provider>
    );
};

describe("JewelleryPage component", () => {
    it("Renders heading correctly", () => {
        renderComponent([]);
        expect(screen.getByText("Jewellery")).toBeInTheDocument();
    });

    it("Renders only jewellery products", () => {
        const products = [
            { id: 1, title: "Necklace", category: "jewelery", price: 100, image: "necklace.jpg" },
            { id: 2, title: "Bracelet", category: "jewelery", price: 50, image: "bracelet.jpg" },
            { id: 3, title: "T-shirt", category: "men's clothing", price: 20, image: "tshirt.jpg" },
        ];

        renderComponent(products);

        expect(screen.getByText("Necklace")).toBeInTheDocument();
        expect(screen.getByText("Bracelet")).toBeInTheDocument();
        expect(screen.queryByText("T-shirt")).not.toBeInTheDocument();
    });

    it("Renders no products when none are jewellery", () => {
        const products = [
            { id: 1, title: "T-shirt", category: "men's clothing", price: 20, image: "tshirt.jpg" },
            { id: 2, title: "Dress", category: "women's clothing", price: 40, image: "dress.jpg" },
        ];

        renderComponent(products);

        expect(screen.queryByText("T-shirt")).not.toBeInTheDocument();
        expect(screen.queryByText("Dress")).not.toBeInTheDocument();
    });

    it("Snapshot testing", () => {
        const products = [
            { id: 1, title: "Necklace", category: "jewelery", price: 100, image: "necklace.jpg" },
            { id: 2, title: "Bracelet", category: "jewelery", price: 50, image: "bracelet.jpg" },
        ];

        const {container} = renderComponent(products);
        expect(container).toMatchSnapshot();
    });
});