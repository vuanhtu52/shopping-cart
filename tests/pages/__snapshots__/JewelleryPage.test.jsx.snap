// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JewelleryPage component > Snapshot testing 1`] = `
<div>
  <div>
    <h2
      class="text-center text-xl font-bold p-4"
    >
      Jewellery
    </h2>
    <div
      class="grid gap-8 justify-items-center grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 p-4"
    >
      <div
        class="flex flex-col min-w-60 w-60 gap-2 border border-gray-300 p-2 rounded-2xl"
      >
        <a
          href="/product/1"
        >
          <img
            alt="Necklace"
            class="object-contain w-full min-h-64 h-64 p-4"
            src="necklace.jpg"
          />
        </a>
        <a
          class="hover:text-[#fb923c]"
          href="/product/1"
        >
          Necklace
        </a>
        <div
          class="font-bold text-lg"
        >
          $
          100
        </div>
        <button
          class="self-start bg-amber-300 p-2 rounded-xl text-sm"
        >
          Add to cart
        </button>
      </div>
      <div
        class="flex flex-col min-w-60 w-60 gap-2 border border-gray-300 p-2 rounded-2xl"
      >
        <a
          href="/product/2"
        >
          <img
            alt="Bracelet"
            class="object-contain w-full min-h-64 h-64 p-4"
            src="bracelet.jpg"
          />
        </a>
        <a
          class="hover:text-[#fb923c]"
          href="/product/2"
        >
          Bracelet
        </a>
        <div
          class="font-bold text-lg"
        >
          $
          50
        </div>
        <button
          class="self-start bg-amber-300 p-2 rounded-xl text-sm"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
